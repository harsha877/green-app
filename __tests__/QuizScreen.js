import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import QuizScreen from '../src/views/quizScreen';

test("Quiz Screen Loading",()=>{
    const push = jest.fn();
   render(<QuizScreen navigation={{ push }}/>)

})

test("Quiz Screen snapshot",()=>{
    const push = jest.fn();
   let QuizComponent=render(<QuizScreen navigation={{ push }} />).toJSON();
   expect(QuizComponent).toMatchSnapshot();
})

describe("Quiz Screen QueryTesting",()=>{
    const push = jest.fn();
    let QuizComponent;
    beforeEach(()=>{
    QuizComponent=render(<QuizScreen navigation={{ push }} />);
    
    })

    test("Testing Button Text",()=>{
        const {getAllByText}=QuizComponent;
        expect(getAllByText("Refresh").length).toBe(1);
     })

     test("Testing Styles",()=>{
        const props=QuizComponent.toJSON();
        let NavigationStyles=props.children[0].props.style;
        expect(NavigationStyles).toStrictEqual({
            justifyContent: "space-between",
		    flexDirection: "row",
        });})

        test("Testing Styles",()=>{
            const props=QuizComponent.toJSON();
            let TextInputStyles=props.children[0].children[0].props.style;
            expect(TextInputStyles).toStrictEqual({
                margin: 10,
		        width: 230,
		        borderWidth: 1,
		        padding: 0,
		        borderRadius: 30,
		        paddingVertical: 10,
		        paddingHorizontal: 10,
		        alignSelf: "center",
            });}) 

    });