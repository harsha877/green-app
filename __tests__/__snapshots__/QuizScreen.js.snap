// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz Screen snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 30,
          "borderWidth": 1,
          "margin": 10,
          "padding": 0,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
          "width": 230,
        }
      }
    >
       
    </TextInput>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#1ca3ec",
            "borderRadius": 50,
            "margin": 10,
            "padding": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "width": 125,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 18,
              "padding": 5,
              "textAlign": "center",
            }
          }
        >
          Refresh
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={Object {}}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    nestedScrollEnabled={true}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
